### PR TITLE
kettle make_json: skip and log rows bigger than 100MB

### LINKED ISSUE #23135: kettle appears to be down
**What happened**:

- go.k8s.io/triage results only extend to july 28th
- triage job is passing https://testgrid.k8s.io/sig-testing-misc#triage&graph-metrics=test-duration-minutes&width=5
- kettle-metrics job is failing https://testgrid.k8s.io/sig-testing-misc#metrics-kettle&width=5
- kettle-staging deployment is crashloopbackoff due to out of disk
- kettle deployment still appears to be up

**What you expected to happen**:

kettle should be up

**How to reproduce it (as minimally and precisely as possible)**:

....

**Please provide links to example occurrences, if any**:

see above

**Anything else we need to know?**:

probably we should have alerting for the kettle-metrics job

### PR DESCRIPTION
**Pull request by matthyx**
fixes #23135 
/assign @BenTheElder

### CODE REVIEW

File: kettle/make_json.py [hunk 1]
```diff
@@ -312,6 +322,11 @@ def main(db, opts, outfile):
 
     rows_emitted = set()
     for rowid, row in make_rows(db, builds):
+        meterfile = MeterFile()
```
Comment (BenTheElder) [inline kettle/make_json.py:325]: this is clever, but does it make more sense to wrap this up in a method that gets the json size and keep the class as an implementation detail?

File: kettle/make_json.py [hunk 2]
```diff
@@ -312,6 +322,11 @@ def main(db, opts, outfile):
...
+        json.dump(row, meterfile, sort_keys=True)
+        if meterfile.size > 104857600:
```
Comment (BenTheElder) [inline kettle/make_json.py:327]: we should explain this constant.

File: kettle/make_json.py [hunk 3]
```diff
@@ -312,6 +322,11 @@ def main(db, opts, outfile):
...
+            print('skipping row of size %d for path %s' % (meterfile.size, row['path']))
```
Comment (BenTheElder) [inline kettle/make_json.py:328]: we should mention that this is because it exceeds the limit

**Review by BenTheElder — COMMENTED** [review]
thank you!

cc @spiffxp

**Comment by matthyx** [comment]
@BenTheElder done

**Review by BenTheElder — APPROVED** [review]
/lgtm
/approve

I think we should try to get kettle unblocked with this, we can revisit later if we should say attempt to truncate the log instead.

**Pull request merged**